src/android/io/branch/BranchSDK.java

```diff
@@ -508,6 +519,60 @@ private void generateShortUrl(int instanceIdx, JSONObject options, JSONObject co
 
     }
 
+    /**
+     * Generate a QR code.
+     *
+     * @param qrCodeSettings   A {@link JSONObject} value to set QR cide options.
+     * @param instanceIdx   The instance index from branchObjects array
+     * @param options       A {@link JSONObject} value to set URL options.
+     * @param controlParams A {@link JSONObject} value to set the URL control parameters.
+     */
+    private void getBranchQRCode(JSONObject qrCodeSettings, int instanceIdx, JSONObject options, JSONObject controlParams, CallbackContext callbackContext) throws JSONException {
+
+        BranchLinkProperties linkProperties = createLinkProperties(options, controlParams);
+
+        BranchUniversalObjectWrapper branchUniversalWrapper = (BranchUniversalObjectWrapper) this.branchObjectWrappers.get(instanceIdx);
+        BranchUniversalObject buo = branchUniversalWrapper.branchUniversalObj;
+
+        BranchQRCode branchQRCode = new BranchQRCode();
+        if (qrCodeSettings.has("codeColor")) branchQRCode.setCodeColor(qrCodeSettings.getString("codeColor"));
```

Comment:
Code style: if statement bodies should always have a new line and be encased in braces. An example of this style is under `createLinkProperties`